superqt/_eliding_label.py

```diff
@@ -0,0 +1,105 @@
+from typing import List
+
+from superqt.qtcompat.QtCore import QPoint, QRect, QSize, Qt
+from superqt.qtcompat.QtGui import QFont, QFontMetrics, QResizeEvent, QTextLayout
+from superqt.qtcompat.QtWidgets import QLabel
+
+
+class QElidingLabel(QLabel):
+    """A QLabel variant that will elide text (add '…') to fit width.
+
+    QElidingLabel(parent: Optional[QWidget] = None, f: Qt.WindowFlags = ...)
+    QElidingLabel(text: str, parent: Optional[QWidget] = None, f: Qt.WindowFlags = ...)
+
+    For a multiline eliding label, use `setWordWrap(True)`.  In this case, text
+    will wrap to fit the width, and only the last line will be elided.
+    When `wordWrap()` is True, `sizeHint()` will return the size required to fit the
+    full text.
+    """
+
+    ELIDE_STRING = "…"
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self._elide_mode = Qt.TextElideMode.ElideRight
+        self._text = ""
+        if args and isinstance(args[0], str):
+            self._text = args[0]
+
+    # New Public methods
+
+    def elideMode(self) -> Qt.TextElideMode:
+        """The current Qt.TextElideMode."""
+        return self._elide_mode
+
+    def setElideMode(self, mode: Qt.TextElideMode):
+        """Set the elide mode to a Qt.TextElideMode."""
+        self._elide_mode = Qt.TextElideMode(mode)  # type: ignore
+        super().setText(self._elidedText())
+
+    def fullText(self) -> str:
+        """Return the full un-elided text."""
+        return self._text
+
+    @staticmethod
+    def wrapText(text, width, font=QFont()) -> List[str]:
```

Comment:
This feels weird `font=QFont()` 

Why not use a `None` ?

Reply:
can do,  (have to convert to an empty font for the call to QTextLayout, but can do `QTextLayout(text, font or QFont())`)